### ISSUE TITLE
Forking a controller

### ISSUE THREAD

**Issue by jnicklas** [thefrontside/effection.js#33]
If I have a function which returns a controller, I currently have to write this:

``` js
fork(function*() { yield returnsController() });
```

It would be nice if there were some syntax sugar which would allow me to write:

``` js
fork(returnsController());
```

With the exact same behaviour as above.

Yay or nay?

**Comment by cowboyd** [comment]
Yes, very much so. It's becoming clearer that there is still something more primitive to uncover. I think that the controller might be it, and that what we really want to do is yield / fork to a controller, and in both cases, the process is resolving the thing into a controller.

In other words, when we see: `yield X`, we need to have a function `f(X) -> Controller` so that we're yielding to a controller. Same with `fork(X)` we can use `f` to convert to a controller and then proceed.

**Comment by cowboyd** [comment]
Closed by #57 

`fork` is now an operation that takes any operation:

* `fork(new Promise())`
* `fork()`
* `fork(fork())`

are all valid forks.
